### ISSUE TITLE
Mautic 3

### ISSUE THREAD

**Issue by madmath03** [mautic/docker-mautic#149]
Is there any plans to upgrade this docker images for Mautic 3 or would you prefer to provide the both of them ?

**Comment by RCheesley** [comment]
@luizeof any thoughts on this?

**Comment by luizeof** [comment]
@RCheesley i'm working on a container to do this ;)

**Comment by rucksman** [comment]
Any progress on this? Although the description on Docker hub suggests that "mautic/mautic:latest" would fetch v3, it still pulls only 2.16.0. Even pulling mautic/mautic:v3.0.0-beta (which is to be found on Docker hub in "Builds") just gives me 2.16.0.

**Comment by rucksman** [comment]
My fault: after cleaning up my volumes at least mautic/mautic:v3.0.0-beta did work.

**Comment by eximius313** [comment]
Shouldn't "latest" be 3.0.0 since it was already released?

**Comment by rucksman** [comment]
Is this still maintained? Or should we rather build our own Dockerfile?

**Comment by gabrielpeixoto** [comment]
I use mautic on the docker on my server.

What happens if I update this inside the system (/s/update)? I did this once from 2.16.2 to 2.16.3.

![image](https://user-images.githubusercontent.com/3699061/86297427-b0fab200-bbd1-11ea-8bbc-4d1c5f40208b.png)

Is there a guide on how to update for docker users?

**Comment by fago** [comment]
I just upgrading it using the usual update script. IT fails related to #164:

> 
> Installed Mautic version:	2.16.3
> PHP version:				7.1.33
> OS:							Linux
> Upgrade type:				CLI
> 
> Doing pre-upgrade checks...
> sh: 1: mysqldump: not found
> One or more errors occurred during pre-upgrade checks: 
> - Your server does not meet the minimum PHP requirements. Mautic requires PHP version 7.2.21 while your server has 7.1.33. Please contact your host to update your PHP installation.

- Your MySQL/MariaDB version is not supported. You need at least MySQL 5.7.14 or MariaDB 10.2 in order to run Mautic 3.

**Comment by ahilles107** [comment]
we have 3.1 already released

**Comment by Bellusterra** [comment]
@ahilles107 docker pull mautic/mautic:v3 pulls 3.0 from docker hub. Does something need to be added/changed to pull 3.1?

**Comment by CodeBradley** [comment]
> @RCheesley i'm working on a container to do this ;)

Any updates on this? I'm interested. Ty

**Comment by loganto** [comment]
@Bellusterra 

Ive been running default mautic/mautic:v3 docker -  and updated to 3.1 from CLI just fine.  
However not sure if this is what the recommended approach is.
  
It went something like this:   
Get into docker: `docker exec mautic bash`

```
root@89976357efb5:/var/www/html# bin/console mautic:update:find
Version 3.1.2 of Mautic is available for download. Please visit https://github.com/mautic/mautic/releases/tag/3.1.2 for more information.
To update, you can run 'php bin/console mautic:update:apply' from the command line.


root@89976357efb5:/var/www/html# php bin/console mautic:update:apply
Are you sure you wish to update Mautic to the latest version? y
Step    5 [----->----------------------] Clearing the cache

<warning>IMPORTANT: Run the same command again with --finish. For example 'php bin/console mautic:update:apply --finish'</warning>


root@89976357efb5:/var/www/html# php bin/console mautic:update:apply --finish 
Step    2 [============================] Success! You are now running version 3.1.2 of Mautic.
```

**Comment by Bellusterra** [comment]
@loganto Thank you, I was not familiar with updating this way. I appreciate you taking the time to respond.

**Comment by loganto** [comment]
@Bellusterra no problems.
I've been interested in what is a proper way to maintain / update dockerized mautic in a long run - so I've reviewed all the opened issues. Found some details in these 2 issues:
```
      https://github.com/mautic/docker-mautic/issues/2
      https://github.com/mautic/docker-mautic/issues/90
```

Per  maintainer:
>>> By the way mautic is developed, I do not advise updating it just by recreating the container. Some caching and database operations are usually required. Mautic also offers a simple script that can be used to upgrade. https://github.com/mautic/mautic/blob/staging/upgrade.php
  
TLDR the way I understood it  (happy to be wrong)- mautic mixes data (state) with code by design - in upstream. Which is, well -  le sigh, hate that - but some apps do that I suppose - like wordpress.   
    
- Thus for a new container we cannot easily decouple data from code.  
- Thus it has only a single volume `/var/www/html` for combined data and code.  
- Thus we cannot create stateless mautic container with just new version of the code.  
- Thus as end consumers we cannot simply pull lets say new version of the mautic, ex **mautic:3.1.2**, just have it run an update in the background and just get a new version.      
    
Basically you cant update by updating container version, have to use either update via UI or cli update:  
https://docs.mautic.org/en/setup/updating-at-command-line

There are some articles online withhelpful tips:
https://mauteam.org/mautic/mautic-admins/mautic-update-how-to-dont-click-that-button/
   
Im just hoping ustream fixes this so it's not wordpress/bedrock 2.0

**Comment by GuidoDr** [comment]
Any update on this topic? mautic/mautic:latest does still pull 2.16.3 

And running the upgrade on the console does also not work because the image of 2.16.2 does have an outdated php version :-(
I could only update there at least to version 2.16.3 with the console tasks.
Upgrade to v3 fails.
```
root@xyz:/var/xyz/mautic# docker-compose exec mautic php app/console mautic:update:apply
Sind Sie sicher, dass Sie Mautic  zur neuesten Version aktualisieren möchten? yes
Step    1 [->--------------------------] Update Information wird geladen...

You can only upgrade to Mautic 3 by running 'php upgrade_v3.php' instead of this command.


root@xyz:/var/xyz/mautic# docker-compose exec mautic php upgrade_v3.php
Welcome to the Mautic 3 upgrade script! Before we start, we'll run some pre-upgrade checks to make sure your system is compatible.
IMPORTANT: you will need to update your cron jobs from app/console/* to bin/console/* after the upgrade. You can already change them now if you want.
Please type 'yes' if you're ready to start: yes

Thank you, continuing...
===== STARTING MAUTIC 3 UPGRADE AT 2020-11-21 20:25:09... =====
Installed Mautic version:       2.16.3
PHP version:                            7.1.33
OS:                                                     Linux
Upgrade type:                           CLI

Doing pre-upgrade checks...
sh: 1: mysqldump: not found
One or more errors occurred during pre-upgrade checks:
- Your server does not meet the minimum PHP requirements. Mautic requires PHP version 7.2.21 while your server has 7.1.33. Please contact your host to update your PHP installation.

- Your MySQL/MariaDB version is not supported. You need at least MySQL 5.7.14 or MariaDB 10.2 in order to run Mautic 3.
```

Actually the database in my installation is already MariaDB 10.5.8 (so that second error does not seem to be correct)

**Comment by loganto** [comment]
@GuidoDr 
per dockerhub page: https://hub.docker.com/r/mautic/mautic

> If you want to pull the latest stable image from Mautic 3 Series on DockerHub:

docker pull mautic/mautic:v3

Note that during the 3.0.x period, the 'mautic/mautic:latest' tag still refers to Mautic 2 for backward compatibility. If you intend to use Mautic 2, use the 'mautic/mautic:v2' tag instead of 'mautic/mautic:latest'.

**Comment by jenshine** [comment]
Hi all, similar issue. Trying to upgrade Mautic in Docker from 2.16.2 to ideally 3. 

After entering into the Mautic container, I ran `php app/console mautic:update:find` and `php bin/console mautic:update:apply`. It confirmed that Mautic 2.16.5 was upgraded successfully, but before continuing to upgrade to version 3.2.4, I tried accessing mautic instance from the browser but was unsuccessful. I got the following message. :(

![image](https://user-images.githubusercontent.com/425569/124726454-5a405a80-df40-11eb-9a94-e5d0fd9944c1.png)

I tried clearing cache by running `php app/console cache:clear`, but that didn't seem to change anything. Also, took a look at /var/www/html/app/config/local.php, but not sure if there is anything that needs to be changed there. 

Any ideas on what broke it and how to fix or further troubleshoot this? 

Thank you!

**Comment by jenshine** [comment]
To upgrade to Mautic 3, I ran upgrade_v3.php. It's asking for PHP upgrade to 7.2. Has anyone tried and successfully upgraded PHP version by creating a Docker container with PHP7.2 and then connecting or linking it to the Mautic container? New to Docker so not having the best results. Thx.

**Comment by cibero42** [comment]
Hey there! I'm one of the new maintainers of this repository. It was neglected for some time but that changes now! Could you please re-test if you are still getting this issue so we could focus our limited time on the issues that are still relevant? If there will be no response we'll close this issue in 1 week. It can always be re-opened later on.

**Comment by cibero42** [comment]
Hello,

We'll move forward by supporting v5 and up. I'll close this issue as it is not applicable anymore. Remember that you can always reopen it if you wish to do so!
